carbites.go

```diff
@@ -29,7 +29,7 @@ func Split(in io.Reader, targetSize int, s Strategy) (Splitter, error) {
 	case Simple:
 		return NewSimpleSplitter(in, targetSize)
 	case Treewalk:
-		return NewTreewalkSplitter(in, targetSize)
+		return nil, fmt.Errorf("treewalk strategy caches the entier CAR, which is not allowed due to memory considerations")
```

Comment:
Can we please allow this? It should be up to the user to determine if they are ok to buffer in memory. I think `io.Reader` isn't necessarily buffered anyway...

Reply:
Right now this code path doesn't get accessed anyway, which is why I removed it. 

In `main.go` we first check for and handle treewalk strategy separately -- https://github.com/alanshaw/go-carbites/blob/main/cmd/main.go#L36

Reply:
Oh i guess you mean because carbites.Split is public. Yeah, that makes sense.